jormungandr: test added on pass midnight

kadhikari opened the pull request:
Relates to https://jira.kisio.org/browse/NAVP-1133

Commit:
jormungandr: test added on pass midnight

Commit:
jormungandr: formatt for black

kadhikari commented:
jenkins test this please

kadhikari commented:
jenkins test this please

Commit:
correction typo

Commit:
test: test on pass midnight towards previous day added

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+
+        # /journeys from B to A with departure at 20120614T180100 and arrival at 20120614T180102
+        journey_query = (
+            "journeys?from={f}&to={to}&data_freshness=base_schedule&"
```
You should probably already do the request in realtime (and so reuse it for later tests)

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+        initial_nb_vehicle_journeys = len(pt_response['vehicle_journeys'])
+        assert initial_nb_vehicle_journeys == 7
+
+        # /journeys from B to A with departure at 20120614T180100 and arrival at 20120614T180102
```
We should also test the same request (in realtime) both the day before and the day after, to check that these are stable and not affected by the disruption (or doubled)

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+        journey_query = (
+            "journeys?from={f}&to={to}&data_freshness=realtime&"
+            "datetime={dt}&_current_datetime={dt}&direct_path=none&walking_speed=0.001".format(
+                f='stop_point:stopC', to='stop_point:stopA', dt='20120614T230000'
```
Please also do the request at `20120614T180000`, maybe add `max_duration_to_pt=0` to force waiting until 00:30.
That way we can check that base and realtimes VJ are not both actives.

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+class TestKirinDelayPassMidnightTowardsPreviousDay(MockKirinDisruptionsFixture):
+    def test_delay_pass_midnight_towards_previous_day(self):
+        """
+        Relates to "test_cots_update_trip_with_advance_pass_midnight_on_first_station" in kirin
```
If you rename the test in Kirin, please update here :wink:

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+class TestKirinDelayPassMidnightTowardsNextDay(MockKirinDisruptionsFixture):
+    def test_delay_pass_midnight_towards_next_day(self):
+        """
+        Relates to "test_cots_update_trip_with_delay_pass_midnight_on_first_station" in kirin
```
:+1: great to keep trace of it.
We must also do the same in kirin, to ensure that mock provided here stays the same as result expected in kirin.

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+
+        # /journeys from B to A with departure at 20120614T080100 and arrival at 20120614T080102
+        journey_query = (
+            "journeys?from={f}&to={to}&data_freshness=base_schedule&"
```
same remark: you can use `realtime` here

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+        # A new disruption with a detour on departure station to have  a pass midnight in advance
+        self.send_mock(
+            "vjB",
+            "20120614",
```
This may be the wrong date... Related to the discussion in Kirin: https://github.com/CanalTP/kirin/pull/242/files#r285073475

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2139,6 +2139,231 @@ def test_add_new_trip_present_in_navitia(self):
+        initial_nb_vehicle_journeys = len(pt_response['vehicle_journeys'])
+        assert initial_nb_vehicle_journeys == 7
+
+        # /journeys from B to A with departure at 20120614T080100 and arrival at 20120614T080102
```
Even more important here to test journeys (maybe also arrivals and departures) the day before and after to check that disruption affects the base-VJ on the good day, and that the RT-VJ we get is on the good day.

pbougue reviewed (commented):
Main question here is on circulation_date and the way it's handled in Kraken... I'm pessimistic about the ability to correctly handle an advance pass-midnight with current code.

Commit:
jormungandr: Tests modified after remarks

Commit:
Format black

benoit-bst commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2353,7 +2443,7 @@ def test_delay_pass_midnight_towards_previous_day(self):
         # /journeys from C to A with departure at 20120614T223000 and arrival at 20120615T080102
         journey_query = (
-            "journeys?from={f}&to={to}&data_freshness=realtime&"
+            "journeys?from={f}&to={to}&data_freshness=realtime&max_duration_to_pt=0&"
```
max_duration_to_pt=0 ?

kadhikari replied:
This parameter is to avoid a section on walk.

Commit:
Modify the test on "anti" pass-midnight

* Disruption on vjA (correction)
* Simplify a bit the journey request
* Disruption one day after (to avoid out-of-validity-period error)
* Check (in realtime) that VJs are correctly managed:
    - check deletion of base-VJ the correct day (and not another day)
    - check add of realtime-VJ the correct day (and not another day)

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2196,6 +2196,370 @@ def test_add_new_trip_present_in_navitia_with_a_shift(self):
+        # A new disruption with a detour on departure station to have a pass midnight in advance
+        self.send_mock(
+            "vjA",
+            "20120615",
```
This should probably be 20120616 (as it's supposed to be GTFS base-VJ's date in GTFS-RT's doc), but then the disruption seems to be rejected by Kraken.

pbougue commented on source/jormungandr/tests/kirin_realtime_tests.py:
```diff
@@ -2196,6 +2196,370 @@ def test_add_new_trip_present_in_navitia_with_a_shift(self):
+        # > vjA circulates the day before
+        response = self.query_region(ba_15T08_journey_query)
+        assert len(response['journeys']) == 1
+        assert response['journeys'][0]['departure_date_time'] == '20120615T080100'
```
Currently it's failing on 20120615: this is the (wrong) one that's disrupted (and of course below on 20120616 it's failing too, as it's not disrupted on the good day).

pbougue reviewed (commented):
Current behavior does not seem correct #TDD
